0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -167,14 +115,36 @@ Content-Type: application/spsp4+json
+    * The `destination_account` MUST be used as the STREAM `destinationAccount`.
+    * The `shared_secret` MUST be decoded from base64 and used as the STREAM `sharedSecret`.
+  
+    It MAY respond with additional information if it is an invoice server or a pull payment server. For more information, see [SPSP Invoices](../0035-spsp-invoices/0035-spsp-invoices.md) and [SPSP Pull Payments](../0036-spsp-pull-payments/0036-spsp-pull-payments.md).
+
+3. The SPSP Client establishes a STREAM connection to the SPSP Server using the SPSP Server's ILP address and shared secret.
+
+### Simple push payment
+
+Given the open STREAM connection, the SPSP Client begins sending ILP packets of value.
+  1. The SPSP Client will adjust their STREAM `sendMax` to reflect the amount they're willing to send.
+  2. The SPSP Server will adjust their STREAM `receiveMax` to reflect the amount they're willing to receive.
+  3. The SPSP Client's and Server's STREAM Modules will move as much value as possible while staying inside these bounds.
+  4. If the SPSP Client reaches their `sendMax`, they end the stream and the connection. If the SPSP Server reaches their `receiveMax`, they will end the stream and the connection.
+
+### Simple data transmission
 
-The sender uses the receiver details to create the STREAM connection:
+Given the open STREAM connection, either the SPSP Client or the Server begins sending ILP packets of data.
 
-* The `destination_account` should be used as the STREAM destinationAccount.
-* The `shared_secret` should be decoded from base64 and used as the STREAM sharedSecret.
-* If present, the `balance.maximum - balance.current` SHOULD be used as the STREAM `receiveMax`.
+This data MUST be [UTF-8](https://en.wikipedia.org/wiki/UTF-8) encoded. The size of the data SHOULD be defined setting STREAM `maxOffset`. Each application built on STREAM and using the principle of data transmission MAY define more restrictive requirements. 
 
-In a UI, the `asset_info` and `receiver_info` objects (if present) can be used for display purposes. These objects can be manipulated by the receiver in any way, so amounts SHOULD be displayed in source units when possible.
+All STREAM parameters - `destinationAccount`, `sendMax`, `receiveMax`, `maxOffset` - are defined in [STREAM's frame encoding](../0029-stream/0029-stream.md#53-frames).
 
-Note that the sender can send as many STREAM payments as they want using the same receiver info. The sender SHOULD query the receiver again once the time indicated in the [`Cache-Control` header](#response-headers) has passed.
+Note that the SPSP Client and Server can send as many STREAM payments and data as they want using the same query response. The SPSP Client SHOULD query the Server again once the time indicated in the [`Cache-Control` header](#response-headers) has passed.
```

Comment:
> The SPSP Client SHOULD query the Server again once the time indicated in the [`Cache-Control` header](#response-headers) has passed.

Is this a good idea? I think we're mixing concepts here. 

The cache-control headers tell you how long the connection parameters are useful for but once the connection is established, the STREAM server may be happy to continue processing packets indefinitely.

As an implementor I'd expect the server to drop the connection when it decides that my session is no longer valid I'd not expect to need to take note of cache-control headers during SPSP and keep a timer running that tracks this expiry once I'm using STREAM to exchange money/data.

Reply:
^ @emschwartz @sharafian

Reply:
That's a good point, we should just say that SPSP queries occurring after the Cache-Control limit should trigger a refetch, rather than saying that it should be retried right away once the Cache-Control has expired